### DISCUSSION TITLE
Environment variables vs flags

### DISCUSSION THREAD

**Discussion by adamk9k** [puma/puma#3053] [Q&A] [answered]
Hey there, I've been reading the code/documentation trying to understand this but I'd like a confirmation:

Using the environment variables 
```
- PUMA_MIN_THREADS 
- PUMA_MAX_THREADS
```
would result in the same if I used the `-t <min>:<max>` flag with Puma, right? Ditto for `-w` and `WEB_CONCURRENCY` env variable. 

The reason why I'm asking is I would like to have the same command in every environment for starting Puma: 

`puma -C config/puma.rb`

but change the thread pool size/worker count with environment variables.

So the two calls would be equivalent:

```shell
PUMA_MIN_THREADS=1 PUMA_MAX_THREADS=16 WEB_CONCURRENCY=2 puma -C config/puma.rb
```

and

```shell
puma -t 1:16 -w2 -C config/puma.rb
```

Is this correct?

**Comment by MSP-Greg** [comment, marked as answer]
Yes.  I believe the precedence is:

**Highest**
CLI
config file
ENV
**Lowest**
